pkg/app/topaz/runtime_resolver.go

```diff
@@ -63,19 +62,24 @@ func NewRuntimeResolver(
 		}
 	}
 
-	if cfg.OPA.Config.Discovery != nil && ctrlf != nil {
+	if cfg.OPA.Config.Discovery != nil {
 		host := os.Getenv("ASERTO_HOSTNAME")
 		if host == "" {
 			if host, err = os.Hostname(); err != nil {
 				host = os.Getenv("HOSTNAME")
 			}
 		}
 		details := strings.Split(*cfg.OPA.Config.Discovery.Resource, "/")
-		cleanupController, err := ctrlf.OnRuntimeStarted(ctx, cfg.OPA.InstanceID, "", details[0],
-
-			details[1], host, func(cmdCtx context.Context, cmd *api.Command) error {
-				return management.HandleCommand(cmdCtx, cmd, sidecarRuntime)
-			})
+		if cfg.ControllerConfig.Server.TenantID == "" {
+			cfg.ControllerConfig.Server.TenantID = cfg.OPA.InstanceID // get the tenant id from the opa instance id config.
+		}
+		ctrl, err := controller.NewController(logger, details[0], host, cfg.ControllerConfig, func(cmdCtx context.Context, cmd *api.Command) error {
```

Comment:
There is no check if `details` contains an element, I so think `details[0]` can panic?

Reply:
Would `len(details) == 0` be more concise to identity the state, as arrays cannot have a negative length.

```
if len(details) == 0 {
			return nil, func() {}, aerr.ErrBadRuntime.Msg("provided discovery resource not formatted correctly")
		}
```